src/prettyprint.lua

```diff
@@ -0,0 +1,229 @@
+local COLOURS = require("src.utils.consoleColor")
+local PASS_SYMBOL = COLOURS.PASS("√")
+local ERROR_SYMBOL = "●"
+local FAIL_SYMBOL = COLOURS.FAIL("×")
+local PASS_HEADER = COLOURS.PASS_HEADER(" PASS ")
+local FAIL_HEADER = COLOURS.FAIL_HEADER(" FAIL ")
+
+-- Matches Jest
+local TAB_SIZE = 2
+
+local NodeType = require("src.interface.testnodetype")
+local tablex = require("src.utils.tablex")
+local prettyValue = require("src.utils.prettyValue")
+
+local function getTabs(amount)
+	return (" "):rep(amount * TAB_SIZE)
+end
+
+--- Recursively traverses the node tree while calling the supplied callback.
+---@param node lest.TestResult|lest.Describe
+---@param onNode fun(node: lest.TestResult|lest.Describe, depth: number):boolean A callback which is called for each Describe and Test in the tree. You can return true to keep going or return false to terminate.
```

Comment:
These types are still being incorrectly used.

`lest.*Result` types are the tables returned by `runTests`. These inherit from `lest.TestResultNode`

`lest.*` types, such as `Describe`, are the tables returned by `findTests`. These inherit from `lest.TestNode`